test/e2e/prometheus_test.go

```diff
@@ -305,6 +305,7 @@ func createK8sAppMonitoring(
 }
 
 func testPromRemoteWriteWithTLS(t *testing.T) {
+	t.Parallel()
```

Comment:
This test alone takes around 10-12 minutes to run. It has 20 subtests each of which takes around one minute. We should see if we there are redundant subtests that we can remove, or look for a way to optimize them.

Reply:
I checked quickly a while ago but my recollection was that all combinations made sense to be tested.

Reply:
sgtm, I'll dive deeper into that test once we merge this PR. Just waiting for @paulfantom to chime as well